src/ast/diagnostic.cpp

```diff
@@ -72,4 +72,16 @@ void Diagnostics::emit(std::ostream& out, Severity s, const Diagnostic& d) const
   }
 }
 
+void Diagnostics::absorb(Diagnostics& other)
```

Comment:
cute name but maybe something more idiomatic? `merge` or `join`?

Reply:
`drain`? I am trying to come up with a way to express that they are being moved from `other` also. I was not happy with absorb, could also flip the relationship, e.g. `move_into`?

Reply:
`move_into` isn't bad or maybe `take` ?

Reply:
I've made it so that you can `move` Diagnostics objects, and added an `add` that requires a moved Diagnostics. This seems like the cleanest semantics.